cartoframes/data/enrichment/enrichment_service.py

```diff
@@ -0,0 +1,136 @@
+import pandas as pd
+import geopandas as gpd
+import uuid
+from collections import defaultdict
+
+from ..dataset.dataset import Dataset
+from ...utils.geom_utils import wkt_to_geojson, geojson_to_wkt
+from ...exceptions import EnrichmentException
+from ...auth import get_default_credentials
+from ..clients import bigquery_client
+
+_ENRICHMENT_ID = 'enrichment_id'
+_WORKING_PROJECT = 'carto-do-customers'
+
+
+def enrich(query_function, **kwargs):
+    credentials = _get_credentials(kwargs['credentials'])
+    user_dataset = credentials.username.replace('-', '_')
+    bq_client = _get_bigquery_client(_WORKING_PROJECT, credentials)
+
+    data_copy = _prepare_data(kwargs['data'], kwargs['data_geom_column'])
+    tablename = _upload_dataframe(bq_client, user_dataset, data_copy, kwargs['data_geom_column'])
+
+    query = _enrichment_query(user_dataset, tablename, query_function, **kwargs)
+
+    return _execute_enrichment(bq_client, query, data_copy, kwargs['data_geom_column'])
+
+
+def _get_credentials(credentials=None):
+    return credentials or get_default_credentials()
+
+
+def _get_bigquery_client(project, credentials):
+    return bigquery_client.BigQueryClient(project, credentials)
+
+
+def _prepare_data(data, data_geom_column):
+    data_copy = __copy_data_and_generate_enrichment_id(data, _ENRICHMENT_ID, data_geom_column)
+    data_copy[data_geom_column] = data_copy[data_geom_column].apply(wkt_to_geojson)
+    return data_copy
+
+
+def _upload_dataframe(bq_client, user_dataset, data_copy, data_geom_column):
+    data_geometry_id_copy = data_copy[[data_geom_column, _ENRICHMENT_ID]]
+    schema = {data_geom_column: 'GEOGRAPHY', _ENRICHMENT_ID: 'INTEGER'}
+
+    id_tablename = uuid.uuid4().hex
+    data_tablename = 'temp_{id}'.format(id=id_tablename)
+
+    bq_client.upload_dataframe(data_geometry_id_copy, schema, data_tablename,
+                               project=_WORKING_PROJECT, dataset=user_dataset, ttl_days=1)
```

Comment:
If you remove TTL stuff in the upload method you have to remove `ttl_days` parameter

Reply:
Right!

Reply:
Done